src/extensions/default/AutoUpdate/main.js

```diff
@@ -394,47 +428,59 @@ define(function (require, exports, module) {
      */
     function _updateProcessHandler(updates) {
 
-        if(!updates) {
-            console.warn("AutoUpdate : updates information not available.");
-            return;
-        }
-        var OS = getPlatformInfo(),
-            checksum,
-            downloadURL,
-            installerName,
-            platforms,
-            latestUpdate;
 
-        latestUpdate = updates[0];
-        platforms = latestUpdate ? latestUpdate.platforms : null;
+        if (checkIfAnotherSessionInProgress()) {
+            UpdateInfoBar.showUpdateBar({
+                type: "error",
+                title: Strings.AUTOUPDATE_ERROR,
+                description: Strings.AUTOUPDATE_IN_PROGRESS
+            });
+        } else {
+            setAutoUpdateSessionInProgress(true);
 
-        if(platforms && platforms[OS]) {
+            if (!updates) {
+                console.warn("AutoUpdate : updates information not available.");
+                return;
+            }
+            var OS = getPlatformInfo(),
+                checksum,
+                downloadURL,
+                installerName,
+                platforms,
+                latestUpdate;
 
-             //If no checksum field is present then we're setting it to 0, just as a safety check,
-            // although ideally this situation should never occur in releases post its introduction.
-            checksum = platforms[OS].checksum ? platforms[OS].checksum : 0,
-            downloadURL = platforms[OS].downloadURL ? platforms[OS].downloadURL : "",
-            installerName = downloadURL ? downloadURL.split("/").pop() : "";
+            latestUpdate = updates[0];
+            platforms = latestUpdate ? latestUpdate.platforms : null;
 
-        } else {
-            // Update not present for current platform
-            return;
-        }
+            if (platforms && platforms[OS]) {
 
-        if (!checksum || !downloadURL || !installerName) {
-            console.warn("AutoUpdate : asset information incorrect for the update");
-            return;
-        }
+                //If no checksum field is present then we're setting it to 0, just as a safety check,
+                // although ideally this situation should never occur in releases post its introduction.
+                checksum = platforms[OS].checksum ? platforms[OS].checksum : 0,
+                downloadURL = platforms[OS].downloadURL ? platforms[OS].downloadURL : "",
+                installerName = downloadURL ? downloadURL.split("/").pop() : "";
 
-         var updateParams = {
-            downloadURL: downloadURL,
-            installerName: installerName,
-            latestBuildNumber: latestUpdate.buildNumber,
-            checksum: checksum
-        };
+            } else {
+                // Update not present for current platform
+                return;
```

Comment:
Do we need to reset value of "AutoUpdate.inProgress" in these failures?

Reply:
Addressed in 036e4a8